examples/caching/README.md

```diff
@@ -0,0 +1 @@
+# Caching
```

Comment:
Should I delete the tutorials for adapters to be deprecated? I was hesitant and thought we should keep them until we actually remove those features.

I created `examples/caching` so that we can direct people to that link (which will display the `README`) and they won't see / peak at the older features to be deprecated.

Reply:
Ahh, sure, as long as it's clear they're deprecated (big letters with a note/link to the right one) then it's fine to hold them around.